src/main/py/ledger/ledger.py

```diff
@@ -0,0 +1,55 @@
+"""
+Module to manage a ledger for an Artwork.
+
+The Ledger class allows us to maintain a history of ownership for an Artwork.
+"""
+
+import hashlib
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Ledger:
+    """Class to manage a Ledger for a single Artwork."""
+
+    def __init__(self) -> None:
+        self.stack = []
+        self.top_of_stack = None
+
+    def add_owner(self, peer):
+        """
+        Add a new owner to the ledger.
+        """
+
+        if not hasattr(peer, "keys"):
+            logger.error("The owner must have a 'keys' attribute.")
+            return
+
+        if self.top_of_stack is not None:
+            previous_hash = self.top_of_stack.digest()
+        else:
+            previous_hash = b""
+
+        new_hash = hashlib.sha256(peer.keys["public"].encode()).digest()
+        combined_hash = previous_hash + new_hash
+        self.top_of_stack = hashlib.sha256(combined_hash)
+        self.stack.append((peer, self.top_of_stack.digest()))
```

Comment:
Optional/Opportunistic: this is correct and works, but it suffers a bit in readability.

I would simplify this to:

```python
    combined_hash = hashlib.sha256()

    if len(self.stack) > 0:
        current_owner = self.stack[-1]
        combined_hash.update(current_owner[1])

    new_owner_hash = hashlib.sha256(peer.keys["public"].encode()).digest()
    combined_hash.update(new_owner_hash)
    self.stack.append( (peer, combined_hash.digest()) )        
```

and remove the `top_of_stack` member.